src/http/request.rs

```diff
@@ -132,6 +135,24 @@ impl<B> fmt::Debug for Request<B> {
     }
 }
 
+#[cfg(feature = "compat")]
+impl From<Request> for http_types::Request<Body> {
+    fn from(req: Request) -> http_types::Request<Body> {
+        let mut builder = http_types::request::Builder::new();
```

Comment:
By using `From` on all the types, we can skip the builder, and use the `http::Request` directly. I say that just because for a conversion like this, performance would be crucial, and so skipping the error checking when we know there's no errors will be noticeable.

Reply:
Done in 5eb16616a081661d3fbae716e8ce83dc6840f99c